modules/vectorfieldvisualization/processors/integrallinevectortomesh.cpp

```diff
@@ -32,55 +32,100 @@
 #include <modules/vectorfieldvisualization/processors/3d/streamlines.h>
 #include <inviwo/core/datastructures/geometry/basicmesh.h>
 #include <inviwo/core/network/networklock.h>
+#include <inviwo/core/util/zip.h>
 
 namespace inviwo {
-namespace {
-struct MetaDataSampler {
-    MetaDataSampler(const IntegralLine &line, std::string name)
-        : hasMetaData_(line.hasMetaData(name)), v(0), dv(0) {
-        if (hasMetaData_) {
-            it = line.getMetaData(name).begin();
-            v = *it;
-        }
-    }
 
-    operator dvec3 &() { return v; }
-    operator const dvec3 &() const { return v; }
+namespace detail {
 
-    dvec3 operator++() {
-        if (hasMetaData_) {
-            it++;
-            v = *it;
-        } else {
-            v += dv;
-        }
-        return v;
-    }
-    dvec3 operator++(int) {
-        auto prev = v;
-        operator++();
-        return prev;
-    }
+template <typename T>
+static double metadataToDouble(const T &t) {
```

Comment:
you probably want length in this name...

Reply:
Hrm, both yes and now. For scalar it just converts it to double while for vec types it uses the length, so it would make sense to have length in the name only for vecs, but they need to have the same name to be used in the dispatch.

Reply:
use norm then that is valid in both cases... to double is just a useless name...